src/api.mjs

```diff
@@ -162,14 +154,54 @@ async function route(message) {
       return { ...message, error: data.errors[0].message };
     }
 
+    return { ...message, results: data };
+  } else if (type === "ipfs") {
+    let { url, gateway, timeout: timeoutFromMsg } = message.options;
+
+    const nativeIPFSPattern = /^(ipfs:\/\/)([^/?#]+)(.*)/;
```

Comment:
I believe this regex could be part of the schema package or not?

Reply:
This can be part of the schema package but we will need it here too. Because we are using it to split the IPFS message into protocol, hash and CID.

Reply:
potentiall we could also use a fork if is-ipfs that spits out the CID